### PR TITLE
feat(button): add s2d variantProperties for Button

### PR DESCRIPTION
**Pull request by bashmish**
Using new S2D feature - automatic variants.

### CODE REVIEW

File: button/stories/button.stories.tsx [hunk 1]
```diff
@@ -11,40 +11,50 @@ export default meta;
...
+  },
+  parameters: {
+    s2d: {
+      variantProperties: [
+        'variant',
+        'compact',
+        'disabled',
+        {
+          fromArg: 'icon',
+          values: [iconNames[0]],
```
Comment (bashmish) [inline button/stories/button.stories.tsx:43]: @gluck @georges-gomes I could't find any way to hide this from the matrix and as a result from the Figma component, this is a subcomponent, therefore it's not needed to be displayed in Figma as a property

image

Comment (bashmish) [reply button/stories/button.stories.tsx:43]: I know I could just define this as a defaultValue on `argTypes`, but I don't think it's good to require again using Storybook definitions to achieve this goal, as the whole point of `s2d.variantProperties` was to decouple them, so IMO you need to add a way to set default argument values in `s2d.variantsProperties` too, maybe even add `s2d.defaultArgValues` or something like that

File: button/stories/button.stories.tsx [hunk 2]
```diff
@@ -11,40 +11,50 @@ export default meta;
...
         },
-      },
+        {
+          fromArg: 'iconPosition',
+          values: [
+            { name: 'no icon', argValue: false }, // TODO: `undefined` (also `null`) do not work, `false` is a workaround
+            { name: 'left icon', argValue: 'left' },
+            { name: 'right icon', argValue: 'right' },
```
Comment (bashmish) [inline button/stories/button.stories.tsx:50]: @gluck isn't it a bug that these are sorted alphabetically when importing to Figma? I expect that the order will be as defined here, but I always get it sorted

image

File: button/stories/button.stories.tsx (hunk 1 above)
Comment (gluck) [reply button/stories/button.stories.tsx:43]: Yes I thought it was unnecessary as you could already define args in the story, but I kind-of agree about the decoupling, will expose initArgs there (already available when writing the variant definition inside the plugin)

File: button/stories/button.stories.tsx (hunk 2 above)
Comment (gluck) [reply button/stories/button.stories.tsx:50]: Not sure where this sorting is done, didn't do anything in particular for it but I agree the order should be retained

File: button/stories/button.stories.tsx (hunk 1 above)
Comment (bashmish) [reply button/stories/button.stories.tsx:43]: I tried this, the icon is not rendered then, practically the default value is ignored



then tried this, all icons make over button 40k variants, logically



tried this for fun, it just breaks



looks like `initArgs` doesn't solve this issue

**Pull request merged**